functorch/functorch/csrc/BatchRulesLinearAlgebra.cpp

```diff
@@ -351,6 +351,34 @@ oneOutput cross_batch_rule(const Tensor& self, c10::optional<int64_t> self_bdim,
   return std::make_tuple(linalg_cross(self_, other_, dim_), 0);
 }
 
+c10::optional<int64_t> batch_dim_if_not_empty(const Tensor& t) {
+  if (t.dim() == 1 && t.size(0) == 0) {
+    return c10::optional<int64_t>();
+  }
+  return c10::optional<int64_t>(0);
+}
+
+fourOutputs linalg_lstsq_batch_rule(
+    const Tensor& self, c10::optional<int64_t> self_bdim, const Tensor& b, c10::optional<int64_t> b_bdim,
+    c10::optional<double> rcond, c10::optional<c10::string_view> driver) {
+  TORCH_CHECK(!self_bdim.has_value() || self.dim() > 2, "torch.linalg.lstsq: input must have at least 2 dimensions.");
+  TORCH_CHECK(!b_bdim.has_value() || b.dim() > 1, "torch.linalg.lstsq: other must have at least 1 dimension.");
```

Comment:
I might be reading the docs wrong (or maybe the docs are wrong?) but it looks like B is required to have 2 or more dimensions in https://pytorch.org/docs/stable/generated/torch.linalg.lstsq.html#torch.linalg.lstsq

Reply:
Ahh we should update the docs. I was matching this
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/BatchLinearAlgebra.cpp#L3624-L3625

And checking locally, this works:
```python
A = torch.randn(3, 4)
b = torch.randn(3)
torch.linalg.lstsq(A, b)
```